src/VerifyExamples/Ast.elm

```diff
@@ -1,107 +1,117 @@
+        Type str ->
+            str
+
+        Function _ str ->
+            str
 
 
-group : List Ast -> { localFunctions : List Ast, imports : List Ast, types : List Ast }
+group : List Ast -> Groupped
 group ast =
-    groupHelp ast { localFunctions = [], imports = [], types = [] }
+    groupHelp ast empty
 
 
-groupHelp :
-    List Ast
-    -> { localFunctions : List Ast, imports : List Ast, types : List Ast }
-    -> { localFunctions : List Ast, imports : List Ast, types : List Ast }
+groupHelp : List Ast -> Groupped -> Groupped
 groupHelp ast acc =
     case ast of
         [] ->
             acc
 
-        (Assertion _) :: rest ->
+        (Assertion assertion) :: (Expectation expectation) :: rest ->
+            { assertion = assertion, expectation = expectation }
+                :: acc.examples
+                |> setExamples acc
+                |> groupHelp rest
+
+        (Assertion assertion) :: rest ->
             groupHelp rest acc
 
         (Expectation str) :: rest ->
             groupHelp rest acc
 
         (Import str) :: rest ->
-            groupHelp rest { acc | imports = acc.imports ++ [ Import str ] }
+            Import str
+                :: acc.imports
+                |> setImports acc
+                |> groupHelp rest
 
         (Type str) :: rest ->
-            groupHelp rest { acc | types = acc.types ++ [ Type str ] }
+            Type str
+                :: acc.types
+                |> setTypes acc
+                |> groupHelp rest
 
-        (LocalFunction name str) :: rest ->
-            groupHelp rest { acc | localFunctions = acc.localFunctions ++ [ LocalFunction name str ] }
+        (Function name str) :: rest ->
+            Function name str
+                :: acc.functions
+                |> setFunctions acc
+                |> groupHelp rest
 
 
-toString : Ast -> String
-toString ast =
-    case ast of
-        Assertion str ->
-            str
+empty : Groupped
+empty =
+    { functions = []
+    , imports = []
+    , types = []
+    , examples = []
+    }
 
-        Expectation str ->
-            str
 
-        Import str ->
-            str
+setExamples : Groupped -> List Example -> Groupped
+setExamples groupped examples =
+    { groupped | examples = examples }
 
-        Type str ->
-            str
 
-        LocalFunction _ str ->
-            str
+setImports : Groupped -> List Ast -> Groupped
+setImports groupped imports =
+    { groupped | imports = imports }
+
+
+setTypes : Groupped -> List Ast -> Groupped
+setTypes groupped types =
+    { groupped | types = types }
 
 
-fromIntermediateAst : List IntermediateAst -> List Ast
-fromIntermediateAst =
-    IAst.convert
-        { maybeExpression = Assertion
-        , arrowPrefixed = Expectation
-        , importPrefixed = Import
-        , typePrefixed = Type
-        , function = LocalFunction
-        }
+setFunctions : Groupped -> List Ast -> Groupped
+setFunctions groupped functions =
+    { groupped | functions = functions }
```

Comment:
Shouldn't these be flipped? i.e groupped last?

Reply:
yes, I know. the pipeline looks so nice though :-)